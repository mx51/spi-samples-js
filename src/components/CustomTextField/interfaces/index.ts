import { IFormEventValue } from '../../PairPage/PairForm/interfaces';

type Partial<T> = {
  [P in keyof T]?: T[P];
};

type InputProps = {
  endAdornment?: React.ReactNode;
};

type TInputProps = {
  maxLength?: number;
};

export interface ICustomTextField {
  disabled?: boolean;
  error?: boolean;
  fullWidth?: boolean;
  helperText?: string;
  id?: string;
  inputProps?: Partial<TInputProps>;
  InputProps?: Partial<InputProps>;
  label?: string;
  margin?: 'dense' | 'none';
  onBlur?: (event: IFormEventValue) => void;
  onChange?: (event: IFormEventValue) => void;
  required?: boolean;
  value?: string;
  variant?: 'filled' | 'standard' | 'outlined' | undefined;
}

export interface IErrorInputAdornment {
  isValid: boolean;
}
