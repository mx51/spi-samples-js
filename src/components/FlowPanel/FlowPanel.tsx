import React from 'react';
import Box from '@material-ui/core/Box';
import Drawer from '@material-ui/core/Drawer';
import Typography from '@material-ui/core/Typography';
import { useAppSelector } from '../../redux/hooks';
import { selectedShowFlowPanel } from '../../redux/reducers/CommonSlice/commonSliceSelectors';
import useStyles from './index.styles';

export default function FlowPanel({ children }: { children: React.ReactNode }): React.ReactElement {
  const classes = useStyles();
  const showFlowPanel = useAppSelector(selectedShowFlowPanel);

  return (
    <Drawer
      anchor="right"
      classes={{
        paper: classes.drawerPaper,
      }}
      className={classes.drawer}
      data-test-id="flowPanel"
      open={showFlowPanel}
      variant="persistent"
    >
      <Box className={classes.flowBoxWrapper}>
        <Box className={classes.flowBox}>
          <Typography variant="h6" component="h1">
            Flow
          </Typography>
          {children}
        </Box>
      </Box>
    </Drawer>
  );
}
