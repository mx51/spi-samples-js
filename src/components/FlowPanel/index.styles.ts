import { makeStyles, Theme, createStyles } from '@material-ui/core/styles';
import { drawerWidth } from '../../definitions/constants/commonConfigs';

export default makeStyles((theme: Theme) =>
  createStyles({
    drawer: {
      width: `${drawerWidth}%`,
      flexShrink: 0,
    },
    drawerPaper: {
      width: `${drawerWidth}%`,
      border: 'none',
      background: 'transparent',
      display: 'flex',
      justifyContent: 'center',
    },
    flowBoxWrapper: {
      height: '100%',
      maxHeight: '100%',
      display: 'flex',
      alignItems: 'center',
      padding: theme.spacing(12, 0, 6),
    },
    flowBox: {
      background: theme.palette.common.white,
      border: `1px solid ${theme.palette.secondary.main}`,
      borderRight: 'none',
      borderTopLeftRadius: theme.spacing(0.5),
      borderBottomLeftRadius: theme.spacing(0.5),
      width: '100%',
      height: '100%',
      overflowY: 'scroll', // handle flow status longer messages
      padding: theme.spacing(2),
      '& > h1': {
        paddingBottom: theme.spacing(2),
      },
    },
    flowContent: {
      marginTop: -theme.spacing(2),
      whiteSpace: 'pre-wrap',
      wordWrap: 'break-word',
    },
  })
);
