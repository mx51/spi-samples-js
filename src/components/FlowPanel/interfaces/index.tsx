export interface IFlowPanel {
  showFlowPanel: boolean;
  receiptToggle?: boolean;
}
