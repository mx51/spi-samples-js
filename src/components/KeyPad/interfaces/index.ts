export interface IKeyPadProps {
  title: string;
  subtitle: string;
  defaultAmount: number;
  onAmountChange: (amount: number) => void;
  open: boolean;
  onClose: () => void;
}
