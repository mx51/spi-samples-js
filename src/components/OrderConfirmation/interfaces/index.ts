export interface IOrderConfirmation {
  title: string;
  pathname: string;
  editSubtotal: boolean;
}

export interface ITitleStrategy {
  Pay: string;
  Cashout: string;
  Refund: string;
}
