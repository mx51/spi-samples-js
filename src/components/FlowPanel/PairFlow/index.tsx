import React from 'react';
import { SPI_PAIR_STATUS } from '../../../definitions/constants/commonConfigs';
import { useAppSelector } from '../../../redux/hooks';
import { selectedAcquireConfirmPairingFlow } from '../../../redux/reducers/CommonSlice/commonSliceSelectors';
import { selectPairFormSerialNumber } from '../../../redux/reducers/PairFormSlice/PairFormSelectors';
import { IPairingFlow, ITerminal, ITerminalProps } from '../../../redux/reducers/TerminalSlice/interfaces';
import { terminalInstance, terminalPairingFlow } from '../../../redux/reducers/TerminalSlice/terminalsSliceSelectors';
import useStyles from '../index.styles';

export default function PairFlow({ terminal }: ITerminal): React.ReactElement {
  const classes = useStyles();
  useAppSelector(selectedAcquireConfirmPairingFlow);
  const pairFormSerialNumber = useAppSelector(selectPairFormSerialNumber);
  const currentTerminal = useAppSelector(terminalInstance(pairFormSerialNumber));
  const currentTerminalPairingFlow = useAppSelector(terminalPairingFlow(pairFormSerialNumber));

  const statusInformation = (spi: ITerminalProps) => `
# ----------- STATUS -----------

# POSID: ${spi?.posId}
# EFTPOS: ${spi?.deviceAddress}
# SPI STATUS: ${spi?.status}
# FLOW: ${spi?.flow}

# ------------------------------
# POS: ${spi?.posVersion ? `v${spi?.posVersion}` : '-'} Spi: ${`${spi?.pluginVersion}`}
  `;

  const pairFlowInformation = (pairingFlow: IPairingFlow) => `
### PAIRING PROCESS UPDATE ###

# ${pairingFlow.Message}
# Finished? ${pairingFlow.Finished}
# Successful? ${pairingFlow.Successful}
# Confirmation Code: ${pairingFlow.ConfirmationCode}
# Waiting Confirm from Eftpos? ${pairingFlow.AwaitingCheckFromEftpos}
# Waiting Confirm from POS? ${pairingFlow.AwaitingCheckFromPos}
  `;

  return (
    <pre className={classes.flowContent}>
      {/* For unpaired terminal (Eg: Terminal Pair page) */}
      {pairFormSerialNumber &&
        currentTerminalPairingFlow &&
        currentTerminal.status !== SPI_PAIR_STATUS.PairedConnected &&
        pairFlowInformation(currentTerminalPairingFlow)}
      {pairFormSerialNumber &&
        currentTerminal &&
        currentTerminal.status === SPI_PAIR_STATUS.PairedConnected &&
        statusInformation(currentTerminal)}
      {/* For already paired terminal (Eg: Terminal Details page) */}
      {!pairFormSerialNumber && terminal && statusInformation(terminal)}
    </pre>
  );
}
