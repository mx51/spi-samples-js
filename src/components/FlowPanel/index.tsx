import React from 'react';
import Box from '@material-ui/core/Box';
import Drawer from '@material-ui/core/Drawer';
import Typography from '@material-ui/core/Typography';
import { useAppSelector } from '../../redux/hooks';
import { selectPairFormSerialNumber } from '../../redux/reducers/PairFormSlice/PairFormSelectors';
import { IPairingFlow, ITerminalProps } from '../../redux/reducers/TerminalSlice/interfaces';
import { terminalInstance } from '../../redux/reducers/TerminalSlice/terminalsSliceSelectors';
import useStyles from './index.styles';
import { IFlowPanelState } from './interfaces';

export default function FlowPanel({ flow, terminal }: IFlowPanelState): React.ReactElement {
  const classes = useStyles();
  const pairFormSerialNumber = useAppSelector(selectPairFormSerialNumber);
  const currentTerminal = useAppSelector(terminalInstance(pairFormSerialNumber)) as ITerminalProps;

  const statusInformation = (spi: Any) => `
# ----------- STATUS -----------

# POSID: ${spi?.posId}
# EFTPOS: ${spi?.deviceAddress}
# SPI STATUS: ${spi?.status}
# FLOW: ${spi?.flow}

# ------------------------------
# POS: ${spi?.posVersion ? `v${spi?.posVersion}` : '-'} Spi: ${`${spi?.pluginVersion}`}
  `;

  const pairFlowInformation = (pairingFlow: IPairingFlow) => `
### PAIRING PROCESS UPDATE ###

# ${pairingFlow.Message}
# Finished? ${pairingFlow.Finished}
# Successful? ${pairingFlow.Successful}
# Confirmation Code: ${pairingFlow.ConfirmationCode}
# Waiting Confirm from Eftpos? ${pairingFlow.AwaitingCheckFromEftpos}
# Waiting Confirm from POS? ${pairingFlow.AwaitingCheckFromPos}
  `;

  return (
    <Drawer
      anchor="right"
      classes={{
        paper: classes.drawerPaper,
      }}
      className={classes.drawer}
      data-test-id="flowPanel"
      open={flow}
      variant="persistent"
    >
      <Box className={classes.flowBoxWrapper}>
        <Box className={classes.flowBox}>
          <Typography variant="h6" component="h1">
            Flow
          </Typography>
          <pre className={classes.flowContent}>
            {currentTerminal?.pairingFlow && pairFlowInformation(currentTerminal?.pairingFlow)}
            {pairFormSerialNumber && currentTerminal && statusInformation(currentTerminal)}
            {/* Above one is for unpaired terminal (Eg: Terminal Pair page) */}
            {!pairFormSerialNumber && terminal && statusInformation(terminal)}
            {/* Above one is for already paired terminal (Eg: Terminal Details page) */}
          </pre>
        </Box>
      </Box>
    </Drawer>
  );
}
