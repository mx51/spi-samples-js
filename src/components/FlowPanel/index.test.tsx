import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { cleanup } from '@testing-library/react';
import FlowPanel from '.';
import { IPairFormParams } from '../../redux/reducers/PairFormSlice/interfaces';
import { ITerminalState } from '../../redux/reducers/TerminalSlice/interfaces';
import mockWithRedux, { defaultMockPairFormParams, pairedMockTerminals } from '../../utils/tests/common';

function setupContainer(
  pairForm: IPairFormParams = defaultMockPairFormParams,
  terminals: ITerminalState = pairedMockTerminals
) {
  const customizedStore = {
    getState: () => ({
      common: { showFLowPanel: true, acquireConfirmPairingFlow: true },
      pairForm,
      terminals,
    }),
    subscribe: jest.fn(),
    dispatch: jest.fn(),
  };

  return mockWithRedux(
    <FlowPanel>
      <h6>Flow</h6>
      <p>Mock flow panel content</p>
    </FlowPanel>,
    customizedStore
  );
}

describe('Test <FlowPanel />', () => {
  let mockContainer: Any;

  beforeEach(() => {
    mockContainer = setupContainer();
  });

  afterEach(cleanup);

  test('should match FlowPanel snapshot test', () => {
    // Assert
    expect(mockContainer).toMatchSnapshot();
  });
});
