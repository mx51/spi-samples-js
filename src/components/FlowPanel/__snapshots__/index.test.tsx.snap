// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <FlowPanel /> should match FlowPanel snapshot test 1`] = `
<div>
  <div
    class="MuiDrawer-root MuiDrawer-docked makeStyles-drawer-1"
    data-test-id="flowPanel"
  >
    <div
      class="MuiPaper-root MuiDrawer-paper makeStyles-drawerPaper-2 MuiDrawer-paperAnchorRight MuiDrawer-paperAnchorDockedRight MuiPaper-elevation0"
      style="visibility: hidden; webkit-transform: translateX(1024px) translateX(0px); transform: translateX(1024px) translateX(0px);"
    >
      <div
        class="MuiBox-root MuiBox-root-6 makeStyles-flowBoxWrapper-3"
      >
        <div
          class="MuiBox-root MuiBox-root-7 makeStyles-flowBox-4"
        >
          <h1
            class="MuiTypography-root MuiTypography-h6"
          >
            Flow
          </h1>
          <h6>
            Flow
          </h6>
          <p>
            Mock flow panel content
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
